Python/optimizer_bytecodes.c

```diff
@@ -217,6 +217,59 @@ dummy_func(void) {
         }
     }
 
+    op(_TO_BOOL, (value -- res)) {
+        (void)value;
+        res = sym_new_type(ctx, &PyBool_Type);
+        OUT_OF_SPACE_IF_NULL(res);
+    }
+
+    op(_TO_BOOL_BOOL, (value -- value)) {
+        if (sym_matches_type(value, &PyBool_Type)) {
+            REPLACE_OP(this_instr, _NOP, 0, 0);
+        }
+        else {
+            sym_set_type(value, &PyBool_Type);
```

Comment:
This will need a bottom check (like in gh-#116089). Also the other `sym_set_...()` calls you added below.